### DISCUSSION TITLE
DataTable not showing up on certain configurations

### DISCUSSION THREAD

**Discussion by JonDehn** [Textualize/textual#4394] [General]
Anybody have a problem with DataTables not being displayed on certain machines?  I have an application that makes use of DataTables, among other widgets.  I've run this on Windows 10 and Windows 11, with python 3.11 and 3.12, with the latest version of textual.  

On one of the machines, there is just a blank area where the datatable should be (no rows and no headers, just blank screen).  When running in dev mode, there are no error messages that show up in the textual console.

The configuration that doesn't work has a display device driver of "Intel UHD Graphics 630".  I've run successfully on laptops with "Intel IRIS XE Graphics" and desktops with Nvidia GEForce cards.  I can only imagine that this has something to do with the graphics card or device driver..  Is there anything special about DataTables in the way that they get displayed on Windows that might explain this?

**Comment by JonDehn** [comment]
PS -- I should have mentioned that I get the same behavior in either Windows Powershell or the Command Prompt terminals.

**Comment by davep** [comment]
Textual simply uses the terminal you run it in. There's nothing about the DataTable widget in particular that would be affected by that sort of hardware (at least nothing under Textual's control).

**Comment by TomJGooding** [comment]
Let's try starting with some basics.

What do you see when running the demo with `py -m textual`?

Could you paste the results from `textual diagnose` (available from the [devtools](https://textual.textualize.io/guide/devtools/))?

**Comment by JonDehn** [comment]
This gets a little stranger...

1) running py -m textual shows a table
2) I realized that my tables were included in a TabPane, so I made a small sample program to display a table within a TabbedContent - TabPane, and that showed the table.  In this case, there were two scroll bars, one for the table and one for the app.  I recall a suggestion about that to make the height of the the table explicitly 100% in the css; I set it to 1fr (thinking this would be the same as 100% for something with just one widget in it), and the second scroll bar disappeared as expected.
3) I then changed the tcss for DataTable in my app to be 'height: 1fr' instead of 100%, and voila, in one case, the table appeared.  I have other tables in other tabbed panes that don't appear, even with a 1fr setting.  So I will continue to play around with that.

In case this sheds some light on the subject, here is the output from textual diagnose:


# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.55.1 |
| Rich    | 13.7.1 |

## Python

| Name           | Value                                                             |
|----------------|-------------------------------------------------------------------|
| Version        | 3.12.2                                                            |
| Implementation | CPython                                                           |
| Compiler       | MSC v.1937 64 bit (AMD64)                                         |
| Executable     | C:\Users\kathl\AppData\Local\Programs\Python\Python312\python.exe |

## Operating System

| Name    | Value      |
|---------|------------|
| System  | Windows    |
| Release | 11         |
| Version | 10.0.22621 |

## Terminal

| Name                 | Value     |
|----------------------|-----------|
| Terminal Application | *Unknown* |
| TERM                 | *Not set* |
| COLORTERM            | *Not set* |
| FORCE_COLOR          | *Not set* |
| NO_COLOR             | *Not set* |

## Rich Console options

| Name           | Value               |
|----------------|---------------------|
| size           | width=79, height=25 |
| legacy_windows | True                |
| min_width      | 1                   |
| max_width      | 79                  |
| is_terminal    | False               |
| encoding       | cp1252              |
| max_height     | 25                  |
| justify        | None                |
| overflow       | None                |
| no_wrap        | False               |
| highlight      | None                |
| markup         | None                |
| height         | None                |

**Reply by TomJGooding** [reply]
Most likely the reason is that you have different versions of Textual installed on your various machines.

I think this _might_ be a problem I introduced by changing the max-height of a DataTable back to 100% unfortuately.... But I haven't had a chance to look into this in any detail yet.

**Reply by TomJGooding** [reply]
From a quick dive into this, strangely a table inside an auto height container does seems to display fine, but not inside a TabbedContent. You can just add a height to the tab pane or table, but I don't yet understand why it doesn't display with the default styling.

```python
from textual.app import App, ComposeResult
from textual.widgets import TabbedContent, DataTable, TabPane
from textual.containers import Container

ROWS = [
    ("lane", "swimmer", "country", "time"),
    (4, "Joseph Schooling", "Singapore", 50.39),
    (2, "Michael Phelps", "United States", 51.14),
    (5, "Chad le Clos", "South Africa", 51.14),
    (6, "László Cseh", "Hungary", 51.14),
    (3, "Li Zhuhao", "China", 51.26),
    (8, "Mehdy Metella", "France", 51.58),
    (7, "Tom Shields", "United States", 51.73),
    (1, "Aleksandr Sadovnikov", "Russia", 51.84),
    (10, "Darren Burns", "Scotland", 51.84),
]


class ExampleApp(App):
    CSS = """
    DataTable {
        max-height: 100%; /* current default */
    }

    TabbedContent {
        border: solid blue;
    }

    TabPane {
        border: solid red;
    }

    #auto-height {
        height: auto;
        border: solid green;
    }
    """

    def compose(self) -> ComposeResult:
        with TabbedContent():
            with TabPane("Table"):
                yield DataTable()

        with Container(id="auto-height"):
            yield DataTable()

    def on_mount(self) -> None:
        tables = self.query(DataTable)
        for table in tables:
            table.add_columns(*ROWS[0])
            table.add_rows(ROWS[1:])


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```
